### ISSUE TITLE
Add new core event to optionally disable the allow-list check if required by a plugin

### ISSUE THREAD

**Issue by enual** [piwik/piwik#20874]
It seems AdvertisingConversionExport does not work when login_allowlist_ip is set.

We have this option active: config/config.ini.php : login_allowlist_ip[]

Now matomo blocks out Microsoft when they retrieve /matomo/index.php?module=AdvertisingConversionExport&action=generateConversionExport.

The error message in /var/log/apache/error.log is seems to be "client denied by server configuration: …/matomo/config/global.ini.php"

I tested this : when removing all IPS from login_allowlist_ip, the Export of AdvertisingConversionExport works.

The URL in question is index.php?module=AdvertisingConversionExport&action=generateConversionExport&accessToken=HBX…

Additional information:

Matomo Version 4.14.2
AdvertisingConversionExport 4.1.3

**Comment by michalkleiner** [comment]
Hi @enual, this looks like a premium plugin issue, please create a PG ticket in JIRA, unless it's been already identified this is a problem with Matomo core. In that case feel free to reopen.

**Comment by AltamashShaikh** [comment]
@michalkleiner This issue needs changes in core and post that only we can add this in the plugin, it requires a new event to allow disabling the allow list check for certain actions

**Comment by michalkleiner** [comment]
Ok, thanks @AltamashShaikh. I found that hard to decipher that from the issue description here as it doesn't mention there's any history to the problem.

**Comment by michalkleiner** [comment]
For more info see internal PG-2870.

**Comment by mattab** [comment]
Does this also impact Google Ads similarly how you mention it impacts Microsoft?

**Comment by sgiehl** [comment]
@mattab yes. That affects all external sources that try to fetch data from Matomo without having their IP on the allowlist
